subprojects/core/src/main/java/org/gradle/internal/service/scopes/BuildScopeServices.java

```diff
@@ -185,6 +186,10 @@ public void execute(ServiceRegistration registration) {
         });
     }
 
+    protected TaskStatistics createTaskStatistics() {
```

Comment:
We should really make this a build tree scoped service.

Also, it would be good to later generalize this into a more reusable build metrics collector, that can collect metrics and then forward them to where ever they need to go, such as a build scan or the console. This will allow us to easily add instrumentation (which we always need to do) and have this instrumentation wired into the tooling we use.